jujugui/static/gui/src/app/utils/changes-utils.js

```diff
@@ -0,0 +1,242 @@
+            changeItem.description = change.command.args[0] + ' unexposed';
+            break;
+          case '_add_relation':
+            var serviceList = this.getRealRelationEndpointNames(
+                change.command.args, services);
+            changeItem.icon = 'changes-relation-added';
+            changeItem.description = change.command.args[0][1].name +
+                ' relation added between ' +
+                serviceList[0] + ' and ' + serviceList[1] + '.';
+            break;
+          case '_remove_relation':
+            changeItem.icon = 'changes-relation-removed';
+            changeItem.description = change.command.args[0][1].name +
+                ' relation removed between ' +
+                change.command.args[0][0] +
+                ' and ' +
+                change.command.args[1][0] + '.';
+            break;
+          case '_addMachines':
+            var machineType = change.command.args[0][0].parentId ?
+                'container' : 'machine';
+            changeItem.icon = 'changes-' + machineType + '-created';
+            changeItem.description = change.command.args[0].length +
+                ' ' + machineType +
+                (change.command.args[0].length !== 1 ? 's have' : ' has') +
+                ' been added.';
+            break;
+          case '_destroyMachines':
+            /*jshint -W004*/
+            var machineType = change.command.args[0][0].indexOf('/') !== -1 ?
+                'container' : 'machine';
+            changeItem.icon = 'changes-' + machineType + '-destroyed';
+            changeItem.description = change.command.args[0].length +
+                ' ' + machineType +
+                (change.command.args[0].length !== 1 ? 's have' : ' has') +
+                ' been destroyed.';
+            break;
+          case '_set_config':
+            var cfgServ = services.getById(change.command.args[0]);
+            changeItem.icon = 'changes-config-changed';
+            changeItem.description = 'Configuration values changed for ' +
+                cfgServ.get('displayName').match(removeBrackets)[1] + '.';
+            break;
+          default:
+            changeItem.icon = 'changes-service-exposed';
+            changeItem.description = 'An unknown change has been made ' +
+                'to this enviroment via the CLI.';
+            break;
+        }
+      }
+      if (skipTime || !change) {
+        changeItem.time = '00:00';
+      } else {
+        changeItem.time = this._formatAMPM(new Date(change.timestamp));
+      }
+      return changeItem;
+    },
+
+    /**
+      Return formatted time for display.
+
+      @method _formatAMPM
+      @param {Date} date The current date.
+    */
+    _formatAMPM: function(date) {
+      var hours = date.getHours();
+      var minutes = date.getMinutes();
+      var ampm = hours >= 12 ? 'pm' : 'am';
+      hours = hours % 12;
+      hours = hours ? hours : 12;
+      minutes = minutes < 10 ? '0' + minutes : minutes;
```

Comment:
It looks like that bit is doing zero padding, so the output would be 00:08, not 00:8.